toolchain/check/convert.cpp

```diff
@@ -153,9 +153,9 @@ static auto MakeElementAccessInst(Context& context, SemIR::LocId loc_id,
     // index so that we don't need an integer literal instruction here, and
     // remove this special case.
     auto index_id = block.template AddInst<SemIR::IntValue>(
-        loc_id,
-        {.type_id = context.GetBuiltinType(SemIR::BuiltinInstKind::IntType),
-         .int_id = context.ints().AddUnsigned(llvm::APInt(32, i))});
+        loc_id, {.type_id = context.GetBuiltinType(
+                     SemIR::BuiltinInstKind::IntLiteralType),
+                 .int_id = context.ints().AddUnsigned(llvm::APInt(64, i))});
```

Comment:
```suggestion
                 .int_id = context.ints().Add(i)});
```

There might be a need to cast here, but TBH I think we shouldn't be supporting access at values that wouldn't fit. And, probably values here are going to fit into an IntId directly, again skipping an APInt conversion.

Maybe the argument type should be changed from size_t? ([which should probably have been size_t instead of std::size_t too](https://google.github.io/styleguide/cppguide.html#Integer_Types)).

Casting version:

```suggestion
                 .int_id = context.ints().Add(static_cast<int64_t>(i))});
```

Reply:
(https://github.com/carbon-language/carbon-lang/pull/4546 for std::)